src/reports.cpp

```diff
@@ -447,17 +447,21 @@ static config unit_abilities(const unit* u, const map_location& loc)
 REPORT_GENERATOR(unit_abilities, rc)
 {
 	const unit *u = get_visible_unit(rc);
+	const team &viewing_team = rc.teams()[rc.screen().viewing_team()];
 	const map_location& mouseover_hex = rc.screen().mouseover_hex();
+	const map_location& displayed_unit_hex = rc.screen().displayed_unit_hex();
+	const map_location& hex = (mouseover_hex.valid() && !viewing_team.shrouded(mouseover_hex)) ? mouseover_hex : displayed_unit_hex;
 
-	return unit_abilities(u, mouseover_hex);
+	return unit_abilities(u, hex);
 }
 REPORT_GENERATOR(selected_unit_abilities, rc)
 {
 	const unit *u = get_selected_unit(rc);
 
 	const map_location& mouseover_hex = rc.screen().mouseover_hex();
 	const unit *visible_unit = get_visible_unit(rc);
-	if(visible_unit && u && visible_unit->id() != u->id() && mouseover_hex.valid())
+	const team &viewing_team = rc.teams()[rc.screen().viewing_team()];
+	if (visible_unit && u && visible_unit->id() != u->id() && mouseover_hex.valid() && !viewing_team.shrouded(mouseover_hex))
```

Comment:
I suppose this shrouded check would need to change too, but even before that, what does this change fix? #4456 doesn't affect the selected_unit_abilities report, does it?

Reply:
It does not, I changed this to be consistent with the `REPORT_GENERATOR(unit_abilities)`.